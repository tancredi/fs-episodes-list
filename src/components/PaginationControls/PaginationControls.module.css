.pagination {
  display: flex;
  align-items: center;
  margin: 60px 0;
  font-size: 16px;
}

.pagination a {
  color: #fff;
  border-radius: 3px;
  padding: 20px 50px;
  border-radius: 3px;
  background-color: #362973;
  text-decoration: none;
  font-size: 16px;
  font-weight: 100;
  transition: background 0.2s;
}

.pagination a:hover {
  background-color: #4d3e96;
}

.current {
  flex: 1;
  color: #fff;
  font-weight: 100;
  text-align: center;
}

@media (max-width: 800px) {
  .pagination {
    flex-direction: column;
    margin-top: 30px;
    margin-bottom: 30px;
  }

  .current {
    margin-bottom: 20px;
  }

  .pagination a {
    order: 1;
    margin-top: 10px;
    width: 100%;
    box-sizing: border-box;
    text-align: center;
  }
}
