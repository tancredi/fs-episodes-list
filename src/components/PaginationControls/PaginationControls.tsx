import Link from "next/link";
import { FunctionComponent } from "react";
import type { PaginationData } from "src/types/pagination";
import styles from "./PaginationControls.module.css";

interface Props {
  pagination: PaginationData;
  baseUrl: string;
}

export const PaginationControls: FunctionComponent<Props> = ({
  pagination: { page, totalPages },
  baseUrl,
}) => (
  <div className={styles.pagination}>
    {page > 0 && (
      <Link href={`${baseUrl}?page=${page - 1}`}>&laquo; Previous</Link>
    )}

    <div className={styles.current}>
      Page {page + 1} of {totalPages}
    </div>

    {page < totalPages - 1 && (
      <Link href={`${baseUrl}?page=${page + 1}`}>Next &raquo;</Link>
    )}
  </div>
);
