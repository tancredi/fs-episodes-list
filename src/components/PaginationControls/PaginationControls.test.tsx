import React from "react";
import { queryByText, render } from "@testing-library/react";
import type { PaginationData } from "../../types/pagination";
import { PaginationControls } from "./PaginationControls";

const MOCK_PAGINATION: PaginationData = {
  page: 1,
  totalPages: 3,
};

describe("Pagination controls component", () => {
  it("renders UI correctly", () => {
    expect(
      render(
        <PaginationControls baseUrl="/" pagination={MOCK_PAGINATION} />
      ).asFragment()
    ).toMatchSnapshot();
  });

  it.each([
    [0, 3, false, true],
    [1, 3, true, true],
  ])(
    "renders correct links depending on current page and page count page: %s, total pages: %s, shows previous: %s, shows next: %s",
    (page, totalPages, shouldRenderPrevious, shouldRenderNext) => {
      const { queryByText } = render(
        <PaginationControls
          pagination={{ ...MOCK_PAGINATION, page, totalPages }}
          baseUrl="/base"
        />
      );

      const previous = queryByText("Previous", { exact: false });
      const next = queryByText("Next", { exact: false });

      shouldRenderPrevious
        ? expect(previous).toBeInTheDocument()
        : expect(previous).not.toBeInTheDocument();

      previous &&
        expect(previous).toHaveAttribute("href", `/base?page=${page - 1}`);

      shouldRenderNext
        ? expect(next).toBeInTheDocument()
        : expect(next).not.toBeInTheDocument();

      next && expect(next).toHaveAttribute("href", `/base?page=${page + 1}`);
    }
  );
});
