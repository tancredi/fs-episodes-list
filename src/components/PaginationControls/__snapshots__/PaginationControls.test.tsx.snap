// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination controls component renders UI correctly 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <a
      href="/?page=0"
    >
      « Previous
    </a>
    <div
      class="current"
    >
      Page 2 of 3
    </div>
    <a
      href="/?page=2"
    >
      Next »
    </a>
  </div>
</DocumentFragment>
`;
